### DISCUSSION TITLE
"Severing Files" Problems

### DISCUSSION THREAD

**Discussion by mahowa** [saltyshiomix/nextron#480] [Ideas]
I had a terrible time trying to get electron-serve into my application. I'm not sure which part of my configuration was causing it to break but no matter what I tried I kept getting a `require() of ES Module` error. And of course if you dont have a way to serve your files in production electron becomes pretty useless.

Anyway if anyone finds themselves in the same boat here is the solutions that worked for me.

Utilizing the electron protocol handler:

Middleware to handle http routing in production
```
import {  protocol} from "electron";

  if (isProd) {
    protocol.handle("app", (req) => {
      const { host, pathname } = new URL(req.url);
      try {
        const basePath = path.join(
          __dirname,
          pathname
        );
        const filePath = `file://${basePath}`;
        return net.fetch(filePath);
      } catch (e) {
        console.error("Issue with middleware", host, pathname);
      }
    });
  }
```

initially load the window.
```
await mainWindow.loadURL(
      `app://./${homePath}`
      )
```

**Comment by threelittle87** [comment]
Fixed by downgrading to electron-serve@1.3.0
